test/pylib/manager_client.py

```diff
@@ -173,6 +174,49 @@ async def after_test(self, test_case_name: str, success: bool) -> None:
         logger.info("Cluster after test %s: %s", test_case_name, cluster_status)
 
         return cluster_status
+    
+    async def check_all_errors(self, exclude_errors: list[str] | None = None) -> str | None:
+        critical_error_pattern = r"Assertion.*failed|AddressSanitizer|Aborting on shard"
+        errors = defaultdict(dict)
+        filter_expr = "|".join(exclude_errors) if exclude_errors else None
+
+        for server in await self.all_servers():
+            log_file = await self.server_open_log(server_id=server.server_id)
+            if found_critical := await log_file.grep(critical_error_pattern, filter_expr=filter_expr):
+                errors[server]["critical"] = [e[0] for e in found_critical]
+            if found_errors := await log_file.grep_for_errors(distinct_errors=True):
+                errors[server]["error"] = found_errors
+        found_cores = await self.find_cores()
+
+        # Build a nicely formatted multiline string
+        lines = []
+        for server, errs in errors.items():
+            lines.append(f">>> Server {server.server_id}")
+            if "critical" in errs:
+                lines.append(f"> Found {len(errs['critical'])} critical error(s):")
+                lines.extend(errs["critical"])
+            if "error" in errs:
+                lines.append(f"> Found {len(errs['error'])} unexpected error(s):")
+                lines.extend(errs["error"])
+            lines.append("")  # blank line between servers
+
+        for core in found_cores:
+            lines.append(f"Found core file: {core}")   
+
+        return "\n".join(lines)
```

Comment:
This is not what I meant. Here you can return the dict with error and core dumps. That later on in the fixture you will be able to use.